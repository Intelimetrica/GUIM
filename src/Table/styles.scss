@import '../styles/colors.scss';

.GUIMTable {
  width: 100%;
  border-collapse: collapse;
  table-layout: fixed;
  & tr {
    height: 100%;
  }
  & .striped tr.trow:nth-child(even) {
    background: var(--light-gray);
  }
  & tr.theader{
    background: var(--white);
  }
  & tr.theader:last-child {
    border-bottom: 2px solid var(--dark-gray);
  }
  & tr.highlighted {
    background: var(--light-blue);
  }
  & th.border{
    border-right: 2px solid #8e8e8e;
  }

  & #ticky {
    display: none;
    width: inherit;
  }
  & .arrowContainer {
    display: inline-flex;
    flex-direction: column;
    align-items: center;
    font-size: 11px;
    & .arrow {
      margin-left: 10px;
      border: solid;
      border-width: 0 1px 1px 0;
      display: inline-block;
      padding: 3px;
      &.asc {
        transform: rotate(-135deg);
        -webkit-transform: rotate(-135deg);
      }
      &.desc {
        transform: rotate(45deg);
        -webkit-transform: rotate(45deg);
      }
    }
    & .sorter {
      color: #bfbfbf;
      display: inline-block;
      font-style: normal;
      line-height: 0;
      text-align: center;
      text-transform: none;
      vertical-align: -.125em;
      text-rendering: optimizeLegibility;
      -webkit-font-smoothing: antialiased;
    }
  }
  & .sortableColumn {
    cursor: pointer;
  }
  & .sorter {
    color: #bfbfbf;
    &.active{
      color: #1890FF;
    }
  }
}
