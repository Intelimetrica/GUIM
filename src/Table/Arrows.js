import React, { Component, Fragment } from "react";
import isEmpty from 'lodash/isEmpty';

import "./styles.scss";

export class Arrows extends Component {
  constructor(props) {
    super(props);
    this.state = {
      arrowUp: false,
      arrowDown: false
    };
  }

  render() {
    let arrowUp = false;
    let arrowDown = false;
    if(this.props.active){
      arrowUp = this.props.order === 'asc';
      arrowDown = this.props.order === 'desc';
    }
    return (
      <div className='arrowContainer'>
        <span role="img" aria-label="caret-up" className={`sorter sorter-up ${arrowUp ? 'active' : ''}`}>
          <svg viewBox="0 0 1024 1024" focusable="false" data-icon="caret-up" width="1em" height="1em" fill="currentColor" aria-hidden="true">
          <path d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z">
          </path>
          </svg>
        </span>
        <span role="img" aria-label="caret-down" className={`sorter sorter-down ${arrowDown ? 'active' : ''}`}>
          <svg viewBox="0 0 1024 1024" focusable="false" data-icon="caret-down" width="1em" height="1em" fill="currentColor" aria-hidden="true">
            <path d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z">
            </path>
          </svg>
        </span>
      </div>
    );
  }
}

export default Arrows;