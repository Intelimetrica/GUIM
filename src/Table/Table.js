import React, { Component, Fragment } from "react";
import "./styles.scss";

export class StickyHeader extends Component {
  constructor(props) {
    super(props);
    this.state = { active: false, width: '100%' };
    this.handleScroll = this.handleScroll.bind(this);
  }

  handleScroll(e) {
    const originalHeaderTop = (this.originalHeader)
      ? this.originalHeader.getBoundingClientRect().top
      : Number.POSITIVE_INFINITY;

    if (originalHeaderTop <= this.props.top && !this.state.active) {
      this.setState({ active: true });
    } else if (originalHeaderTop > this.props.top && this.state.active) {
      this.setState({ active: false });
    }
  }

  componentDidMount() {
    window.addEventListener('scroll', this.handleScroll);
    this.originalHeader = document.getElementById(this.props.id);
    let zIndex = parseInt(this.originalHeader.style.zIndex) || 0;
    zIndex += 1;

    this.setState({
      width: this.originalHeader.getBoundingClientRect().width,
      zIndex
    });
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.handleScroll);
  }

  render() {
    const style = (this.state.active) ? {
      top: this.props.top,
      display: 'block',
      position: 'fixed',
      width: this.state.width,
      zIndex: this.state.zIndex
    } : {}
    return (
      <div id="ticky" style={style}>
        <table className={`GUIMTable ${this.props.className}`}>
          <thead>
            {this.props.headers.map((rowHeader, index_row) => (
              <tr id={`header-${index_row}`} className={`theader `} key={`header-${index_row}`}>
                {rowHeader.map((colHeader, index_col) => {
                  let options = {};
                  if (colHeader.rowSpan) {
                    options.rowSpan = colHeader.rowSpan;
                  }
                  if (colHeader.colSpan) {
                    options.colSpan = colHeader.colSpan;
                  }
                  return (
                    <th className={colHeader.className} key={`row-header-${index_row}-${index_col}`} {...options} >{colHeader.text}</th>)
                }
                )
                }
              </tr>
            ))}
          </thead>
        </table>
      </div>
    )
  }
}
const Table = props => {
  const { body, headers, head_id } = props;
  const sticky_header = (props.sticky_header.active) ? (
    <StickyHeader
      id={head_id}
      headers={headers}
      top={props.sticky_header.top}
      className={props.className}
    />
  ) : '';

  return (
    <Fragment>
      {sticky_header}
      <table className={`GUIMTable ${props.className}`}>
        <thead>
          {headers.map((rowHeader, index_row) => (
            <tr id={`header-${index_row}`} className={`theader `} key={`header-${index_row}`}>
              {rowHeader.map((colHeader, index_col) => {
                let options = {};
                if (colHeader.rowSpan) {
                  options.rowSpan = colHeader.rowSpan;
                }
                if (colHeader.colSpan) {
                  options.colSpan = colHeader.colSpan;
                }
                return (
                  <th className={colHeader.className} key={`row-header-${index_row}-${index_col}`} {...options} >{colHeader.text}</th>)
              }
              )
              }
            </tr>
          ))}
        </thead>
        <tbody className={`${props.striped ? "striped" : ""}`}>
          {
            body.map((el, index) => (
              <tr key={`row-${index}`}
                className={`trow ${(props.row_hovered === index) ? 'highlighted' : ''} ${props.row_className}`}
                onMouseEnter={props.row_mouseEnter.bind(this, index)}
                onMouseLeave={props.row_mouseLeave.bind(this, index)}>
                {el.map((td, i) => <td key={`td-${index}-${i}`}>{td}</td>)}
              </tr>
            ))
          }
        </tbody>
      </table>
    </Fragment>
  );
}

//TODO: Implement themes
Table.defaultProps = {
  striped: false,
  sticky_header: {
    active: false,
    top: 0
  },
  head_id: `head-${Math.round(Math.random() * 10000)}`,
  headers: [
    [
      {
        text: '',
        className: 'border'
      },
      {
        text: 'Name',
        colSpan: 2,
        className: 'border'
      },
      {
        text: 'Data',
        colSpan: 2
      }
    ],
    [
      {
        text: 'Number',
        rowSpan: 3,
        className: 'border'
      },
      {
        text: 'First Name',
      },
      {
        text: 'Last Name',
        className: 'border'
      },
      {
        text: 'Email'
      },
      {
        text: 'Actions'
      }
    ]
  ],
  body: [
    ["1", "John", "Doe", "john@doe.com", "View - Edit"],
    ["2", "Jane", "Doe", "jane@doe.com", "View - Edit"],
    ["3", "Josue", "Doe", "josue@doe.com", "View - Edit"]
  ],
  row_mouseEnter: (i) => { },
  row_mouseLeave: (i) => { },
  className: "",
  row_className: "",
  row_hovered: -1, //  This one will set the given row's className to highlighted
  theme: "light"
};

export default Table;
