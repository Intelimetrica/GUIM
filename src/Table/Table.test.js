import React from 'react';
import Table, { StickyHeader } from './Table';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import { findDOMNode } from 'react-dom';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

Math.random = jest.fn(() => 0.1234);

describe('<Table />', () => {
  let component;
  let highlighted = -1;
  let tree;

  const onMouseEnterRow = (i) => highlighted = i;
  const onMouseLeaveRow = () => highlighted = -1;

  const refreshTree = () => tree = component.toJSON();
  const reRender = (striped = false,
    ticky = {active: false, top: 0}) => {
      component = renderer.create(
        <Table
          striped={striped}
          sticky_header={ticky}
          head_id="head_test"
          headers={[
            [
              {
                text: 'Products',
                colSpan: 4
              }
            ],
            [
              {
                text: 'Item',
                colSpan: 2,
                className: 'border'
              },
              {
                text: 'Description',
                colSpan: 2,
                className: 'border'
              }
            ]
          ]}
          body={[
            [1, "Wheat", "Good", "200 Bags"],
            [2, "Rice", "Regular", "300 Bags"],
            [3, "Sugar", "Bad", "100 Bags"],
            [4, "Meat", "Good", "200 Kgs"]
          ]}
          row_mouseEnter={onMouseEnterRow}
          row_mouseLeave={onMouseLeaveRow}
          row_hovered={highlighted}
        />
      )
      refreshTree();
    };
  it("match dom snapshot", () => {
    reRender();
    expect(tree).toMatchSnapshot();
  });

  it("highlights row when hover", () => {
    reRender();

    // Mouse enter on row
    tree[1]
      .children[1]
      .children[2].props.onMouseEnter(2);

    reRender();
    expect(tree).toMatchSnapshot();

    // Mouse leave row
    tree[1]
      .children[1]
      .children[2].props.onMouseLeave();

    reRender();
    expect(tree).toMatchSnapshot();
  });

  it("handles striped and non striped rows", () => {
    reRender(true);
    expect(tree).toMatchSnapshot();

    reRender(false);
    expect(tree).toMatchSnapshot();
  });

});

describe('<Table />', () => {
  let component;
  // This div thing is for the getElementById in componentDidMount
  // it is used to assign a width for the sticky header
  let div = document.createElement('div');
  div.setAttribute("id", "head-id");
  document.body.appendChild(div);

  it('handles a sticky header yo', () => {
    component = shallow(
      <StickyHeader
        id={'head-id'}
        headers={[
          [
            {
              text: 'Products',
              colSpan: 4
            }
          ],
          [
            {
              text: 'Item',
              colSpan: 2,
              className: 'border'
            },
            {
              text: 'Description',
              colSpan: 2,
              className: 'border'
            }
          ]
        ]}
        top={0}
      />
    )

    expect(component).toMatchSnapshot();
    component.setState({active: true});
    expect(component).toMatchSnapshot();
  });
});

