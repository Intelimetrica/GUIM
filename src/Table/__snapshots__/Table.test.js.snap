// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> handles a sticky header yo 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StickyHeader
    headers={
      Array [
        "Product",
        "Product Name",
        "Product Quality",
        "Product Quantity",
      ]
    }
    id="head-id"
    top={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <table
        className="GUIMTable undefined"
      >
        <thead>
          <tr
            className="theader"
          >
            <th>
              Product
            </th>
            <th>
              Product Name
            </th>
            <th>
              Product Quality
            </th>
            <th>
              Product Quantity
            </th>
          </tr>
        </thead>
      </table>,
      "id": "ticky",
      "style": Object {},
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <thead>
          <tr
            className="theader"
          >
            <th>
              Product
            </th>
            <th>
              Product Name
            </th>
            <th>
              Product Quality
            </th>
            <th>
              Product Quantity
            </th>
          </tr>
        </thead>,
        "className": "GUIMTable undefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr
            className="theader"
          >
            <th>
              Product
            </th>
            <th>
              Product Name
            </th>
            <th>
              Product Quality
            </th>
            <th>
              Product Quantity
            </th>
          </tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th>
                Product
              </th>,
              <th>
                Product Name
              </th>,
              <th>
                Product Quality
              </th>,
              <th>
                Product Quantity
              </th>,
            ],
            "className": "theader",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "header-0",
              "nodeType": "host",
              "props": Object {
                "children": "Product",
              },
              "ref": null,
              "rendered": "Product",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": "header-1",
              "nodeType": "host",
              "props": Object {
                "children": "Product Name",
              },
              "ref": null,
              "rendered": "Product Name",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": "header-2",
              "nodeType": "host",
              "props": Object {
                "children": "Product Quality",
              },
              "ref": null,
              "rendered": "Product Quality",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": "header-3",
              "nodeType": "host",
              "props": Object {
                "children": "Product Quantity",
              },
              "ref": null,
              "rendered": "Product Quantity",
              "type": "th",
            },
          ],
          "type": "tr",
        },
        "type": "thead",
      },
      "type": "table",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <table
          className="GUIMTable undefined"
        >
          <thead>
            <tr
              className="theader"
            >
              <th>
                Product
              </th>
              <th>
                Product Name
              </th>
              <th>
                Product Quality
              </th>
              <th>
                Product Quantity
              </th>
            </tr>
          </thead>
        </table>,
        "id": "ticky",
        "style": Object {},
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <thead>
            <tr
              className="theader"
            >
              <th>
                Product
              </th>
              <th>
                Product Name
              </th>
              <th>
                Product Quality
              </th>
              <th>
                Product Quantity
              </th>
            </tr>
          </thead>,
          "className": "GUIMTable undefined",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr
              className="theader"
            >
              <th>
                Product
              </th>
              <th>
                Product Name
              </th>
              <th>
                Product Quality
              </th>
              <th>
                Product Quantity
              </th>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  Product
                </th>,
                <th>
                  Product Name
                </th>,
                <th>
                  Product Quality
                </th>,
                <th>
                  Product Quantity
                </th>,
              ],
              "className": "theader",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "header-0",
                "nodeType": "host",
                "props": Object {
                  "children": "Product",
                },
                "ref": null,
                "rendered": "Product",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": "header-1",
                "nodeType": "host",
                "props": Object {
                  "children": "Product Name",
                },
                "ref": null,
                "rendered": "Product Name",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": "header-2",
                "nodeType": "host",
                "props": Object {
                  "children": "Product Quality",
                },
                "ref": null,
                "rendered": "Product Quality",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": "header-3",
                "nodeType": "host",
                "props": Object {
                  "children": "Product Quantity",
                },
                "ref": null,
                "rendered": "Product Quantity",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        "type": "table",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`<Table /> handles a sticky header yo 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StickyHeader
    headers={
      Array [
        "Product",
        "Product Name",
        "Product Quality",
        "Product Quantity",
      ]
    }
    id="head-id"
    top={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <table
        className="GUIMTable undefined"
      >
        <thead>
          <tr
            className="theader"
          >
            <th>
              Product
            </th>
            <th>
              Product Name
            </th>
            <th>
              Product Quality
            </th>
            <th>
              Product Quantity
            </th>
          </tr>
        </thead>
      </table>,
      "id": "ticky",
      "style": Object {
        "display": "block",
        "position": "fixed",
        "top": 0,
        "width": 0,
        "zIndex": 1,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <thead>
          <tr
            className="theader"
          >
            <th>
              Product
            </th>
            <th>
              Product Name
            </th>
            <th>
              Product Quality
            </th>
            <th>
              Product Quantity
            </th>
          </tr>
        </thead>,
        "className": "GUIMTable undefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr
            className="theader"
          >
            <th>
              Product
            </th>
            <th>
              Product Name
            </th>
            <th>
              Product Quality
            </th>
            <th>
              Product Quantity
            </th>
          </tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th>
                Product
              </th>,
              <th>
                Product Name
              </th>,
              <th>
                Product Quality
              </th>,
              <th>
                Product Quantity
              </th>,
            ],
            "className": "theader",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "header-0",
              "nodeType": "host",
              "props": Object {
                "children": "Product",
              },
              "ref": null,
              "rendered": "Product",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": "header-1",
              "nodeType": "host",
              "props": Object {
                "children": "Product Name",
              },
              "ref": null,
              "rendered": "Product Name",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": "header-2",
              "nodeType": "host",
              "props": Object {
                "children": "Product Quality",
              },
              "ref": null,
              "rendered": "Product Quality",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": "header-3",
              "nodeType": "host",
              "props": Object {
                "children": "Product Quantity",
              },
              "ref": null,
              "rendered": "Product Quantity",
              "type": "th",
            },
          ],
          "type": "tr",
        },
        "type": "thead",
      },
      "type": "table",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <table
          className="GUIMTable undefined"
        >
          <thead>
            <tr
              className="theader"
            >
              <th>
                Product
              </th>
              <th>
                Product Name
              </th>
              <th>
                Product Quality
              </th>
              <th>
                Product Quantity
              </th>
            </tr>
          </thead>
        </table>,
        "id": "ticky",
        "style": Object {
          "display": "block",
          "position": "fixed",
          "top": 0,
          "width": 0,
          "zIndex": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <thead>
            <tr
              className="theader"
            >
              <th>
                Product
              </th>
              <th>
                Product Name
              </th>
              <th>
                Product Quality
              </th>
              <th>
                Product Quantity
              </th>
            </tr>
          </thead>,
          "className": "GUIMTable undefined",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr
              className="theader"
            >
              <th>
                Product
              </th>
              <th>
                Product Name
              </th>
              <th>
                Product Quality
              </th>
              <th>
                Product Quantity
              </th>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  Product
                </th>,
                <th>
                  Product Name
                </th>,
                <th>
                  Product Quality
                </th>,
                <th>
                  Product Quantity
                </th>,
              ],
              "className": "theader",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "header-0",
                "nodeType": "host",
                "props": Object {
                  "children": "Product",
                },
                "ref": null,
                "rendered": "Product",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": "header-1",
                "nodeType": "host",
                "props": Object {
                  "children": "Product Name",
                },
                "ref": null,
                "rendered": "Product Name",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": "header-2",
                "nodeType": "host",
                "props": Object {
                  "children": "Product Quality",
                },
                "ref": null,
                "rendered": "Product Quality",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": "header-3",
                "nodeType": "host",
                "props": Object {
                  "children": "Product Quantity",
                },
                "ref": null,
                "rendered": "Product Quantity",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        "type": "table",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`<Table /> handles striped and non striped rows 1`] = `
Array [
  "",
  <table
    className="GUIMTable "
  >
    <thead>
      <tr
        className="theader "
        id="head_test"
      >
        <th>
          Product
        </th>
        <th>
          Product Name
        </th>
        <th>
          Product Quality
        </th>
        <th>
          Product Quantity
        </th>
      </tr>
    </thead>
    <tbody
      className="striped"
    >
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          1
        </td>
        <td>
          Wheat
        </td>
        <td>
          Good
        </td>
        <td>
          200 Bags
        </td>
      </tr>
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          2
        </td>
        <td>
          Rice
        </td>
        <td>
          Regular
        </td>
        <td>
          300 Bags
        </td>
      </tr>
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          3
        </td>
        <td>
          Sugar
        </td>
        <td>
          Bad
        </td>
        <td>
          100 Bags
        </td>
      </tr>
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          4
        </td>
        <td>
          Meat
        </td>
        <td>
          Good
        </td>
        <td>
          200 Kgs
        </td>
      </tr>
    </tbody>
  </table>,
]
`;

exports[`<Table /> handles striped and non striped rows 2`] = `
Array [
  "",
  <table
    className="GUIMTable "
  >
    <thead>
      <tr
        className="theader "
        id="head_test"
      >
        <th>
          Product
        </th>
        <th>
          Product Name
        </th>
        <th>
          Product Quality
        </th>
        <th>
          Product Quantity
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          1
        </td>
        <td>
          Wheat
        </td>
        <td>
          Good
        </td>
        <td>
          200 Bags
        </td>
      </tr>
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          2
        </td>
        <td>
          Rice
        </td>
        <td>
          Regular
        </td>
        <td>
          300 Bags
        </td>
      </tr>
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          3
        </td>
        <td>
          Sugar
        </td>
        <td>
          Bad
        </td>
        <td>
          100 Bags
        </td>
      </tr>
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          4
        </td>
        <td>
          Meat
        </td>
        <td>
          Good
        </td>
        <td>
          200 Kgs
        </td>
      </tr>
    </tbody>
  </table>,
]
`;

exports[`<Table /> highlights row when hover 1`] = `
Array [
  "",
  <table
    className="GUIMTable "
  >
    <thead>
      <tr
        className="theader "
        id="head_test"
      >
        <th>
          Product
        </th>
        <th>
          Product Name
        </th>
        <th>
          Product Quality
        </th>
        <th>
          Product Quantity
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          1
        </td>
        <td>
          Wheat
        </td>
        <td>
          Good
        </td>
        <td>
          200 Bags
        </td>
      </tr>
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          2
        </td>
        <td>
          Rice
        </td>
        <td>
          Regular
        </td>
        <td>
          300 Bags
        </td>
      </tr>
      <tr
        className="trow highlighted "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          3
        </td>
        <td>
          Sugar
        </td>
        <td>
          Bad
        </td>
        <td>
          100 Bags
        </td>
      </tr>
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          4
        </td>
        <td>
          Meat
        </td>
        <td>
          Good
        </td>
        <td>
          200 Kgs
        </td>
      </tr>
    </tbody>
  </table>,
]
`;

exports[`<Table /> highlights row when hover 2`] = `
Array [
  "",
  <table
    className="GUIMTable "
  >
    <thead>
      <tr
        className="theader "
        id="head_test"
      >
        <th>
          Product
        </th>
        <th>
          Product Name
        </th>
        <th>
          Product Quality
        </th>
        <th>
          Product Quantity
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          1
        </td>
        <td>
          Wheat
        </td>
        <td>
          Good
        </td>
        <td>
          200 Bags
        </td>
      </tr>
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          2
        </td>
        <td>
          Rice
        </td>
        <td>
          Regular
        </td>
        <td>
          300 Bags
        </td>
      </tr>
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          3
        </td>
        <td>
          Sugar
        </td>
        <td>
          Bad
        </td>
        <td>
          100 Bags
        </td>
      </tr>
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          4
        </td>
        <td>
          Meat
        </td>
        <td>
          Good
        </td>
        <td>
          200 Kgs
        </td>
      </tr>
    </tbody>
  </table>,
]
`;

exports[`<Table /> match dom snapshot 1`] = `
Array [
  "",
  <table
    className="GUIMTable "
  >
    <thead>
      <tr
        className="theader "
        id="head_test"
      >
        <th>
          Product
        </th>
        <th>
          Product Name
        </th>
        <th>
          Product Quality
        </th>
        <th>
          Product Quantity
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          1
        </td>
        <td>
          Wheat
        </td>
        <td>
          Good
        </td>
        <td>
          200 Bags
        </td>
      </tr>
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          2
        </td>
        <td>
          Rice
        </td>
        <td>
          Regular
        </td>
        <td>
          300 Bags
        </td>
      </tr>
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          3
        </td>
        <td>
          Sugar
        </td>
        <td>
          Bad
        </td>
        <td>
          100 Bags
        </td>
      </tr>
      <tr
        className="trow  "
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          4
        </td>
        <td>
          Meat
        </td>
        <td>
          Good
        </td>
        <td>
          200 Kgs
        </td>
      </tr>
    </tbody>
  </table>,
]
`;
